### ISSUE TITLE
Software repositories disabled and no applications detected

### ISSUE THREAD

**Issue by pablofsf** [Cogitri/gnome-software-plugin-apk#13]
The software repositories list shows all the software repositories, but the switch next to it is disabled. For each of the repos, it says: "0 applications installed". Looking at the code in gnome software, it looks like for apps to be detected as to belong to a repository, they have to be in the "related" section.
```
gs_app_add_related(repo, app);
```

I believe this was never working, so this is just a tracking issue.

**Comment by Cogitri** [comment]
Hm, the switch did work for me with GS-38 I think, but if we can fix it with GS-40 we should use `gs_app_add_related`.

**Comment by pablofsf** [comment]
Yes, the switches also work for me in GS-38, although the feeling is that as they do not have any apps, toggling the switches is a `noop`. Maybe should be a follow-up issue.

**Comment by pablofsf** [comment]
It looks like there are some additional functions related to the repositories that were added in [GNOME 41](https://gitlab.gnome.org/GNOME/gnome-software/-/merge_requests/812): install, remove, enable and disable repo. I guess there are lots of chances that this is the reason for the visual regression (toggles not being switchable).

Still, as toggling the switches didn't really change anything before, at least this is not a functional regression, I believe.

**Comment by Cogitri** [comment]
Toggling the switches did remove the repository from /etc/apk/repositories, but I doubt anyone actually used that since it’s not very useful.

**Comment by pablofsf** [comment]
Well, I guess then I'll try to figure this out as a priority. Better not to introduce new bugs on upgrade :)
